e2e/test/iothub/service/ConfigurationsClientE2ETests.cs

```diff
@@ -217,5 +212,83 @@ await FluentActions
                 }
             }
         }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task ConfigurationsClient_ClearLabels()
+        {
+            // arrange
+
+            bool configCreated = false;
+            string configurationId = (_idPrefix + Guid.NewGuid()).ToLower(); // Configuration Id characters must be all lower-case.
+            using var sc = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString);
+
+            try
+            {
+                var expected = new Configuration(configurationId)
+                {
+                    Priority = 3,
+                    Labels = { { "labelName", "labelValue" } },
+                    TargetCondition = "deviceId='fakeDevice'",
+                };
+                Configuration addResult = await sc.Configurations.CreateAsync(expected).ConfigureAwait(false);
+                // Labels gets a default dictionary instance, and we want to make sure that if we clear them out, the empty
+                // list gets serialized and the labels on the service-side object indeed get removed.
+                addResult.Labels.Clear();
+
+                // act
+                Configuration updateResult = await sc.Configurations.SetAsync(addResult, false).ConfigureAwait(false);
+
+                // assert
+                updateResult.Labels.Should().BeEmpty();
+            }
+            finally
+            {
+                if (configCreated)
+                {
+                    // If this fails, we shall let it throw an exception and fail the test
+                    await sc.Configurations.DeleteAsync(configurationId).ConfigureAwait(false);
+                }
+            }
+        }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task ConfigurationsClient_NullOutLabels()
+        {
+            // arrange
+
```

Comment:
nit: do we want this extra blank line?

Reply:
I usually include a blank line when the comment refers to a section of statements.

An adjacent comment usually indicates it is relevant only to the next line.